web/packages/teleport/src/Roles/RoleEditor/StandardEditor.tsx

```diff
@@ -168,22 +239,60 @@ const Section = ({
   title,
   tooltip,
   children,
-}: React.PropsWithChildren<{ title: string; tooltip?: string }>) => {
+  removable,
+  onRemove,
+}: React.PropsWithChildren<{
+  title: string;
+  tooltip?: string;
+  removable?: boolean;
+  onRemove?(): void;
+}>) => {
   const theme = useTheme();
   const [expanded, setExpanded] = useState(true);
   const ExpandIcon = expanded ? Icon.Minus : Icon.Plus;
+  const expandTooltip = expanded ? 'Collapse' : 'Expand';
   return (
     <Box
+      as="section"
       border={1}
       borderColor={theme.colors.interactive.tonal.neutral[0].background}
       borderRadius={2}
     >
-      <Flex p={3} css={'cursor: pointer'} onClick={() => setExpanded(e => !e)}>
-        <Flex flex="1" gap={2}>
+      <Flex height="56px" alignItems="center" ml={3} mr={2}>
+        <Flex
```

Comment:
@gzdunek There already is a button that plays the same role (the plus/minus on the right), and this is just for convenience. Won't it be confusing to have two keyboard-accessible elements with the same purpose? Or should I just make it unfocusable?

Reply:
I think we could solve this in an accessible way by using this HTML element https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details. I guess you could style the expand/hide triangle to look like +/- icon?
OTOH I'm not sure if the +/- icon is the best choice here, it look more like add/remove, not expand/hide.

Reply:
@gzdunek I used the `details` and `summary` elements, they work quite well. I also noticed that I forgot that elements that don't render don't count towards validation (it works now, since they are just hidden), so this is now fixed (pending actually showing the validation result, when the section is collapsed, I'll get back to it).

Reply:
One caveat is that I'm handling the state manually, as I will want to customize the state-specific appearance more than CSS allows me to.

Reply:
> I used the details and summary elements, they work quite well.

Awesome!